pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -994,13 +994,18 @@ func (s *sqlEntityServer) watchInit(ctx context.Context, r *entity.EntityWatchRe
 	return lastRv, nil
 }
 
-func (s *sqlEntityServer) poller(stream chan *entity.EntityWatchResponse) {
+func (s *sqlEntityServer) poller(stream chan<- *entity.EntityWatchResponse) {
 	var err error
 
+	// FIXME: we need a way to state startup of server from a (Group, Resource)
+	// standpoint, and consider that new (Group, Resource) may be added to
+	// `kind_version`, so we should probably also poll for changes in there
 	since := int64(0)
+
 	interval := 1 * time.Second
 
 	t := time.NewTicker(interval)
+	defer close(stream)
```

Comment:
Here we close the channel that the broadcaster created for us. This is because we have the send side of the channel. This way it's clear who owns what part of the channel and the associated responsibilities.